amethyst_gltf/src/format/mod.rs

```diff
@@ -0,0 +1,545 @@
+use std::fmt;
+use std::sync::Arc;
+
+use self::importer::{get_image_data, import, Buffers, ImageFormat};
+use assets::{BoxedErr, Format, Source};
+use core::transform::LocalTransform;
+use gfx::Primitive;
+use gfx::texture::SamplerInfo;
+use gltf;
+use gltf::Gltf;
+use itertools::Itertools;
+use renderer::vertex::*;
+
+use super::*;
+
+mod importer;
+
+/// Gltf scene format, will cause the whole default scene to be loaded from the given file.
+///
+/// Using the `GltfSceneLoaderSystem` a `Handle<GltfSceneAsset>` from this format can be attached
+/// to an entity in ECS, and the system will then load the full scene using the given entity
+/// as the root node of the scene hierarchy.
+pub struct GltfSceneFormat;
+
+/// Format errors
+#[derive(Debug)]
+pub enum GltfError {
+    /// Importer failed to load the json file
+    GltfImporterError(self::importer::Error),
+
+    /// GLTF have no default scene and the number of scenes is not 1
+    InvalidSceneGltf(usize),
+
+    /// GLTF primitive use a primitive type not support by gfx
+    PrimitiveMissingInGfx(String),
+
+    /// GLTF primitive missing positions
+    MissingPositions,
+
+    /// External file failed loading
+    LoadError(BoxedErr),
+}
+
+impl StdError for GltfError {
+    fn description(&self) -> &str {
+        use self::GltfError::*;
+        match *self {
+            GltfImporterError(_) => "Gltf import error",
+            InvalidSceneGltf(_) => "Gltf has no default scene, and the number of scenes is not 1",
+            PrimitiveMissingInGfx(_) => "Primitive missing in gfx",
+            MissingPositions => "Primitive missing positions",
+            LoadError(_) => "File loading error",
+        }
+    }
+
+    fn cause(&self) -> Option<&StdError> {
+        match *self {
+            GltfError::GltfImporterError(ref err) => Some(err),
+            GltfError::LoadError(ref err) => Some(err),
+            _ => None,
+        }
+    }
+}
+
+impl fmt::Display for GltfError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::GltfError::*;
+        use std::error::Error;
+        match *self {
+            GltfImporterError(ref err) => {
+                write!(f, "{}: {}", self.description(), err.description())
+            }
+            PrimitiveMissingInGfx(ref err) => write!(f, "{}: {}", self.description(), err),
+            LoadError(ref err) => write!(f, "{}: {}", self.description(), err.description()),
+            InvalidSceneGltf(size) => write!(f, "{}: {}", self.description(), size),
+            MissingPositions => write!(f, "{}", self.description()),
+        }
+    }
+}
+
+impl From<self::importer::Error> for GltfError {
+    fn from(err: self::importer::Error) -> Self {
+        GltfError::GltfImporterError(err)
+    }
+}
+
+impl From<BoxedErr> for GltfError {
+    fn from(err: BoxedErr) -> Self {
+        GltfError::LoadError(err)
+    }
+}
+
+impl Format<GltfSceneAsset> for GltfSceneFormat {
+    const NAME: &'static str = "GLTF";
```

Comment:
Also `.glb` for binary glTF - not sure if you still distinguish by extension.

Reply:
This is just the name of the format, not related to extension at all.